src/cljs/analysis_viewer/fxs.cljs

```diff
@@ -21,28 +21,29 @@
      [:stop {:offset "100%" :stop-color :yellow}]]]
 
    ;; map background
-   [:rect {:x "0" :y "0" :width "100%" :height "100%" :fill (:background-color views/theme)}]
+   [:rect {:x "0" :y "0" :width "100%" :height "100%" :fill (:background-color map-opts)}]
    
    ;; map and data svg
    [:g {}
     [:svg {:view-box "0 0 360 180"}
      ;; map group
      [:g {}
-      (binding [svg-renderer/*theme* views/theme]
-        (svg-renderer/geojson->svg geo-json-map))]
+      (svg-renderer/geojson->svg geo-json-map map-opts)]
 
      ;; data group
      [:g {}
       (when analysis-data
         [:g {}
          (for [primitive-object analysis-data]
            ^{:key (str (:id primitive-object))}
-           (views/map-primitive-object primitive-object 1 time))])]]]])
+           (views/map-primitive-object primitive-object 1 time ui-params))])]]]])
 
 (re-frame/reg-fx
  :spread/download-current-map-as-svg
  (fn [{:keys [geo-json-map analysis-data time]}]
-   (let [ svg-text (html (data-map geo-json-map analysis-data time))
+   (let [params @(re-frame/subscribe [:ui/parameters])
```

Comment:
I know you should definitely not mix event handlers and subs (and there is no need to) but is that fine, a sub in an effect handler? You could wrap it in an event handler that reads from the app-db and passes that to the fx handler.